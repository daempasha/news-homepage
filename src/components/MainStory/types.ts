export interface MainStoryProps {
    title?: string;
    description?: string;
    image?: string
}