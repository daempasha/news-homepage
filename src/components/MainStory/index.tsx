import Button from "@components/Button";
import { MainStoryProps } from "./types";

const MainStory: React.FC<MainStoryProps> = ({ title, image, description }) => (
    <div className="flex-0 grid grid-cols-1 md:grid-cols-2 gap-5">
        <div className="image md:col-span-2">
            <img src={image} alt={title} />
        </div>
        <div className="title text-5xl font-bold row-span-2">{title}</div>
        <div className="description text-gray-800">{description}</div>
        <div className="button"><Button children="READ MORE" /></div>
    </div>
)

export default MainStory;