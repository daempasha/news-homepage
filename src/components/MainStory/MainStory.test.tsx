import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import MainStory from '.';
import { resizeWindow } from '@helper';





describe('<MainStory />', () => {
    it('renders correctly on desktop', () => {
        const mainStory = (<MainStory title="Article title" description="Article description" image={""} />)
        expect(mainStory).toMatchSnapshot()

    });
    it('renders correctly on mobile', () => {
        resizeWindow(640, 1080)
        const mainStory = (<MainStory title="Article title - mobile" description="Article description mobile" image={""} />)
        expect(mainStory).toMatchSnapshot()


    });

});