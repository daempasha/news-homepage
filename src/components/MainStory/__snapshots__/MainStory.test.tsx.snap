// Vitest Snapshot v1

exports[`<MainStory /> > renders correctly on desktop 1`] = `
<MainStory
  description="Article description"
  image=""
  title="Article title"
/>
`;

exports[`<MainStory /> > renders correctly on mobile 1`] = `
<MainStory
  description="Article description mobile"
  image=""
  title="Article title - mobile"
/>
`;
